src/lib.rs

```diff
@@ -13,21 +16,93 @@ fn get_user_input(prompt: &str) -> String {
 }
 
 fn take_guess() -> char {
-    return get_user_input("Provide the character you want to guess: ")
+    // take a single character as guess input
+    return take_user_input("Provide the character you want to guess: ")
         .chars()
         .next()
         .expect("More than one character was guessed.");
 }
 
+const HANGMAN_STAGES: [&str; 7] = [
+    "
+       +---+
+       |   |
+           |
+           |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+           |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+       |   |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|   |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|\\  |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|\\  |
+      /    |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|\\  |
+      / \\  |
+           |
+     =========",
+];
+
+fn print_hangman_stage(incorrect_guesses: usize) {
+    if incorrect_guesses < HANGMAN_STAGES.len() {
+        println!("{}", HANGMAN_STAGES[incorrect_guesses]);
+    } else {
+        panic!("Invalid number of incorrect guesses.");
+    }
```

Comment:
This introduces an  implicit coupling between the number of lives a player has in the `config` and the number of drawing stages. Currently this is shown by the fact that you check whether lives/incorrect guesses do not exceed a magic number both here and in lines 186 - 188.

What do you think about being more "permissive" and simply always draw the last hangman stage if `incorrect_guesses > `HANGMAN_STAGES.len()`? Alternatively, if this coupling is really desired, you could think about making it more explicit by getting rid of `config.lives` and simply determining whether the player is dead by checking if they reached the final hangman stage.